### ISSUE TITLE
Create a class NetworkService that specifically handles the send async requests and recieve answers by REST API

### ISSUE THREAD

**Issue by iBYarchitect** [ios-course/ironfoudation-team-project#122]
## Depends on tasks:
- #120
- https://github.com/ios-course/simple-network-service/issues/3
- https://github.com/ios-course/simple-network-service/issues/5

## This task is a part of:
- #59

## Subtasks: 
- #203.

### Dev requirements and restrictions:

Possible APIs:
``` swift
func sendRequest(request: Request) async -> Result<Data, NetworkError> {}
```
or
```swift
func sendRequest(request: Request, completionHandler: Result<Data, NetworkError> -> Void) {}
```
   - Use URLSession to send request
   - Handle HTTP errors (based on [HTTP status](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes))
   - Handle case there's no internet connection (that means we can handle timeouts, the [timeoutintervalforrequest](https://developer.apple.com/documentation/foundation/urlsessionconfiguration/1408259-timeoutintervalforrequest) might be helpful)

**Comment by dersim-davaod** [comment]
Closed as the duplication of the https://github.com/ios-course/simple-network-service/issues/6

**Comment by kirpichonok** [comment]
Closed with PR [#204](https://github.com/ios-course/ironfoudation-team-project/pull/204).
